### DISCUSSION TITLE
Better Daily Weathercodes

### DISCUSSION THREAD

**Discussion by brennyD** [open-meteo/open-meteo#368] [Ideas]
First of all, this API rocks! It's awesome that something open source and accessible for weather data exists in the age of weather API mafias charging absurd prices for low traffic.

I use this for a weather forecast display in my home, and something that bothers me is the sometimes inaccurate representation of the weather code when calling the daily aggregation API. If a day is completely sunny, with a majority of the hours' weathercode reflecting that, except for one hour where it is cloudy the daily weather code will often opt to return cloudy, even though this isn't the case. A workaround for this would be to get all hours of a given day and do a frequency count of the different weathercodes and display that one, but can't this be done at the API level? 

I am not super versed in swift but on [this line](https://github.com/open-meteo/open-meteo/blob/main/Sources/App/Controllers/ForecastapiController.swift#L556) in the forecast api it looks like a max is being taken of the weather code? Does this mean the max int value of the weathercodes of the day? That would explain the bias for bad weather since those have a higher int value than good weather. Is it possible to make the change to be a frequency counter?

**Comment by patrick-zippenfenig** [comment]
Hi @brennyD,
thanks for your feedback!

Yes, for daily aggregations the API is taking the maximum weather code. A higher weather code indicates more severe weather conditions. It can be argued that returning a weather code with rain can be confusing for the user if there is rain only at night but sunny weather throughout the day. #84 could improve the system by calculating the weather code for morning, afternoon, evening and night.

Using the most frequent weather code can lead to misunderstandings. For instance, during thunderstorms, the severe conditions typically last for only 1-3 hours. You can find more discussion about this topic here https://github.com/open-meteo/open-meteo/issues/228
